Add tutorial Jupyter notebooks

jisoo-lee-furiosaAI opened the pull request:
changes:

- add "notebooks/HowToUseWarboyFromStartToFinish.ipynb" for Warboy tutorial
- add "notebooks/HowToUseWarboyVisionModels.ipynb" for our project tutorial

jisoo-lee-furiosaAI committed:
[add] initial hands-on

jisoo-lee-furiosaAI committed:
[mod] Mask URLs in ipynb notebooks

- Improve readability of markdown cells

jisoo-lee-furiosaAI committed:
[fix] hands-on config files

- typo in demo.yaml file
- modify num_calibration_data in yolov8n.yaml file

jisoo-lee-furiosaAI committed:
[mod] explanation about editing yolo models

- more specific explanation

jisoo-lee-furiosaAI committed:
[mod] explanation about quantization

- more detailed explanation

jisoo-lee-furiosaAI committed:
[mod] refactor HowToUseWarboyFromStartToFinish

- Extracted code from functions
- Rearranged cell order
- Edited markdown cells for better explanation
- Organized imported modules by isort

jisoo-lee-furiosaAI committed:
[mod] prerequisties in HowToUseWarboyVisionModels

- same as HowToUseWarboyFromStartToFinish

jisoo-lee-furiosaAI committed:
[move] relocate notebooks & tutorial config files

- to notebooks/ dir
- keep project root clean and better organized
- notebooks were updated to set the working directory to the root

jisoo-lee-furiosaAI committed:
[fix] HowToUseWarboyFromStartToFinish var

- num_calibration_data to 100

jisoo-lee-furiosaAI committed:
[add] instruction to build decoders

jisoo-lee-furiosaAI committed:
[fix] notebooks/HowToUseWarboyFromStartToFinish

- fixed onnx_i8_path

jisoo-lee-furiosaAI committed:
[fix] notebooks/HowToUseWarboyVisionModels

- cfg paths

jisoo-lee-furiosaAI committed:
[mod] refactor ipynb notebooks

- add 'Set Root Path' and force to set root path
- remove bash and pip cells

jisoo-lee-furiosaAI committed:
[add] instruction about port fowarding

- ensure to run command at local machine

jisoo-lee-furiosaAI committed:
[fix] notebooks/HowToUseWarboyVisionModels

- remove pip install cell

hayeong-lee-furiosa commented:
~~1. HowToUseWarboyFromStartToFinish.ipynb 파일의 NPU Profiling section~~

~~- There can be `OpenTelemetry trace error occurred. cannot send span to the batch span processor because the channel is full` warning messages when writing the trace file. But you can ignore them.~~
~~- 발생 가능한 에러 설명 시 에러의 원인과 해결방법을 조금 더 자세히 설명해주는건 어떨까요?~~
~~- 예시) While writing the trace files, `OpenTelemetry trace error occurred. cannot send span to the batch span processor because the channel is full` error might occur. In this case, wait until any jobs running on NPU is done and try again.~~

2. HowToUseWarboyVisionModels 파일의 End to End performance test section

- `from warboy_vision_models.tests.e2e.test_object_det import test_warboy_yolo_accuracy_det'
- 'test_warboy_yolo_accuracy_det(model_name = 'yolov8n',  model = param['onnx_i8_path'],  input_shape=param['input_shape'],  anchors= param['anchors'])`
- 해당 코드의 test_warboy_yolo_accuracy_det함수 실행시, assert문 충족하지 못해 fail나는것 보다 print문으로 대체해서 목표 수치로 부터 얼마나 떨어져있는지 log로 확인하면 좋을거같다는 의견 드립니다!

3. HowToUseWarboyVisionModels 파일의 Web Demo with Fast API 섹션

- API 섹션 들어가는게 좋아보입니다! 다만, run_web_demo함수 호출시 demo_cfg_path = 'notebooks/demo.yaml' 파일을 찾지 못합니다
- `FileNotFoundError: [Errno 2] No such file or directory: 'notebooks/demo.yaml'`에러 수정 부탁드립니다!

jisoo-lee-furiosaAI committed:
[fix] root path setting

- fix to use Path().resolve().parent instead of os.getcwd()
- fix paths in demo.yaml

jisoo-lee-furiosaAI committed:
[fix] setting root path

- to find current notebook path, use 'os.path.abspath("")'

jisoo-lee-furiosaAI commented:
@hayeong-lee-furiosa 

1. 실제로 에러가 아닌 warning이어서 해결할 필요 없이 trace 파일이 잘 생성되기 때문에 수정하지 않았습니다.
2. 이미 존재하는 e2e-tests 코드를 불러오는 부분이어서 어쩔 수 없이 error가 뜨긴 하는데.. 일괄적으로 assert문의 목표 수치를 95%가 아닌 90%로 조절하는 것은 어떻게 생각하실까요? 코드를 복잡하게 하는 것보단 target 수치를 낮추는 게 나을 것 같아 의견 구합니다.

~~3. 확인해보니, 두 개의 jupyter notebook을 연달아서 실행하면서 kernel을 재시작 해주지 않으면 root path를 설정할 때 꼬이는 오류가 있는 것 같아 일단은 한 번 수정해보았습니다. 저도 테스트를 좀 더 해봐야 할 것 같은데, 같이 한 번 확인해주시면 감사하겠습니다!~~
- 여러번 실행해보니 여전히 좀 이상한 것 같네요... 다시 수정해야 할 것 같습니다.

hayeong-lee-furiosa commented:
2. 넵 좋습니다

jisoo-lee-furiosaAI committed:
[mod] add warning before set root path

- error occurs due to os.chdir(ROOT)
- cannot solve this problem, so warn users to restart kernel or make
sure to run the cell only once

jisoo-lee-furiosaAI committed:
[mod] e2e-tests assertion

- modified target accuracy as 90% (95% is too high)
- modified assertion format in test_instance_seg.py same as others

jisoo-lee-furiosaAI commented:
2번에 대해서 target mAP를 90%로 조정해서 yolov8n object detection 모델은 assertion이 나지 않는 상태입니다.

3번의 경우 해결책이 없는 것 같아 사용자에게 충분히 경고하고, 문제가 생길 경우 커널 재시작을 해야 함을 설명해주는 것으로 우선 결정했습니다. 혹시 좋은 방법 있으면 말씀해주시면 감사하겠습니다.

jisoo-lee-furiosaAI committed:
[mod] preprocessor explanation

- adjusted markdown cell positions for better flow

hayeong-lee-furiosa commented:
LGTM!
